olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/CatalogClient.h

```diff
@@ -59,43 +79,47 @@ class DATASERVICE_READ_API CatalogClient final {
   ~CatalogClient();
 
   /**
-   * @brief Cancel currently active requests.
+   * @brief Cancels the currently active requests.
    * @return True on success
    */
   bool CancelPendingRequests();
 
   /**
-   * @brief fetches catalog configuration asynchronously.
-   * @param request contains the complete set of request parameters.
-   * @param callback will be invoked once the catalog configuration is
-   * available, or an error is encountered.
+   * @brief Fetches the catalog configuration asynchronously.
+   * @param request The `CatalogRequest` instance that contains a complete set
+   * of request parameters.
+   * @param callback The `CatalogResponseCallback` object that is invoked if
+   * the catalog configuration is available or an error is encountered.
    * @return A token that can be used to cancel this request
    */
   client::CancellationToken GetCatalog(CatalogRequest request,
                                        CatalogResponseCallback callback);
 
   /**
-   * @brief fetches catalog configuration asynchronously.
-   * @param request contains the complete set of request parameters.
+   * @brief Fetches the catalog configuration asynchronously.
+   * @param request The `CatalogRequest` instance that contains a complete set
+   * of request parameters.
    * @return CancellableFuture of type CatalogResponse, which when complete will
    * contain the catalog configuration or an error. Alternatively, the
    * CancellableFuture can be used to cancel this request.
```

Comment:
Can we use the following description instead?
```
* @return `CancellableFuture` that contains the `CatalogResponse`
* instance or an error. You can also use `CancellableFuture` to cancel this
* request.
```

Reply:
I don't think it's completely correct. The CatalogResponse contains the error not the CancellableFuture. How about:

`CancellableFuture` that contains the `CatalogResponse` instance with the catalog configuration or an error. You can also use `CancellableFuture` to cancel this request.

Reply:
Agree